### PR TITLE
Fixes #31965 - make bmc_credentials_accessible disabled by default

### PR DESCRIPTION
**Pull request by lzap**
Also remove the settings check as that would be pretty useless with this new value.

### CODE REVIEW

**Review by tbrisker — DISMISSED** [review]
Makes sense to me, this would need an upgrade warning in the manual for folks who rely on this value being available in the enc or templates

**Comment by laugmanuel** [comment]
One interesting find: If I disable the setting (set to `no`), Foreman is no longer able to retrieve the power status.
The API (`/api/hosts/xyz.example.com/power`) returns:
```json
{
  "id": 29172,
  "state": "na",
  "title": "N/A",
  "statusText": "Power operations are not enabled on this host."
}
```
This also results in the UI not showing the BMC-Tab and Power-Button on the host itself.
With the setting enabled (set to `yes`), I get:
```json
{
  "id": 29172,
  "state": "on",
  "title": "On"
}
```
The environment I tested this on is still Foreman 1.24.3. Not sure if something changed since then?

**Comment by tbrisker** [comment]
@laugmanuel that's odd, https://github.com/theforeman/foreman/blob/develop/app/models/nic/bmc.rb#L24 seems to use the unredacted password (and has been already in 1.24)

**Comment by laugmanuel** [comment]
> @laugmanuel that's odd, https://github.com/theforeman/foreman/blob/develop/app/models/nic/bmc.rb#L24 seems to use the unredacted password (and has been already in 1.24)

@tbrisker, hmm. I can reproduce this 100% of the time. The relevant BMC smartproxy is not Foreman itself but an external smartproxy; maybe this does change things? I've also applied the API fix from #8349.

**Comment by tbrisker** [comment]
@laugmanuel that is quite concerning, it means if we merge this PR we will break BMC power status. I'm afraid I don't have a working BMC setup to test with, is it possible you could try to debug where the power state uses the redacted password instead of the unredacted one? I couldn't find any such from reading the code. Also, which BMC provider does your proxy use? that might also help identify the issue.

**Comment by laugmanuel** [comment]
@tbrisker, sure - I can try to locate the relevant code. I'll report back when I find something. I've validated this with the freeipmi provider.

**Comment by laugmanuel** [comment]
@tbrisker, I could verify the same behaviour with Foreman 2.3 (as expected).
The error is caused by this line: https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L794

**Comment by tbrisker** [comment]
Great find @laugmanuel !! 
@lzap can you please update that line so power status is available even with this setting turned off? 

Although that makes me wonder if anyone ever turned it off, considering that this code has been there for at least 8 years and no-one complained about power status not being available. Perhaps we should just get rid of this setting and remove it from enc and template output?

**Comment by lzap** [comment]
I agree, this makes little sense. Foreman is not a password vault, there are many better solutions to the problem.

So the plan is to:

* Remove the setting.
* Change the BMC.password to always return uncrypted password.
* Remove password method from safemode.
* Remove password from YAML output (can someone point where do we actually do this?).

**Comment by lzap** [comment]
Someone explain me this code:

```ruby
    alias_method :password_unredacted, :password
    def password
      Setting[:bmc_credentials_accessible] ? password_unredacted : nil
    end
```

So `password_unredacted` is an alias to `password` and then in `password` method we call the alias. That should be an endless loop? I am not getting this.

**Comment by lzap** [comment]
I rebased.

**Comment by ekohl** [comment]
I'm guessing that in the past the workflow was that people used `ipmitool user set password` to set the password using the configuration management. That's probably why it was in the ENC. So removing it is a breaking change. If we do decide to do this, it must have a deprecation warning and an upgrade warning.

**Comment by lzap** [comment]
Deprecation warning where exactly?

**Comment by tbrisker** [comment]
> Deprecation warning where exactly?

here:
https://github.com/theforeman/theforeman.org/blob/gh-pages/_includes/manuals/nightly/1.2_release_notes.md#upgrade-warnings

**Comment by lzap** [comment]
Oh I thought you mean warning in the code, sure: https://github.com/theforeman/theforeman.org/pull/1850

Rebased.

**Comment by ekohl** [comment]
So we missed 3.0 with this but did merge the upgrade warning in https://github.com/theforeman/theforeman.org/pull/1850. How do we want to deal with this?

**Comment by ezr-ondrej** [comment]
What is the status here? If there will be no activity for two more weeks, we will close, but you're always welcome to reopen.

File: app/registries/foreman/settings/auth.rb [hunk 1]
```diff
@@ -131,12 +131,8 @@
     setting('bmc_credentials_accessible',
       type: :boolean,
       description: N_("Permits access to BMC interface passwords through ENC YAML output and in templates"),
-      default: true,
+      default: false,
       full_name: N_('BMC credentials access'))
-    validates('bmc_credentials_accessible',
-      ->(value) { value || Setting[:safemode_render] },
-      message: N_("Unable to disable bmc_credentials_accessible when safemode_render is disabled"))
```
Comment (ezr-ondrej) [inline app/registries/foreman/settings/auth.rb:133]: This removal doesn't make much sense unless we remove the validation from `safemode_render` as well IMHO.
Also this allows entering the invalid state if users didn't change this setting, but had changed the `safemode_render`

**Review by ezr-ondrej — COMMENTED** [review]
I've rebased for you, but I've got issue with removing just one side of the validations.

**Comment by ezr-ondrej** [comment]
@lzap ? :)

**Comment by lzap** [comment]
STOP IT! ;)

**Comment by lzap** [comment]
How about this. For testing, you can enable BMC SSH module on proxy and set all SSH commands to /usr/bin/true on localhost, that will create a fake BMC that always responds ok but does nothing. It was meant for debugging.

I will yank the upgrade note and prepare a new one for nightly.

**Comment by lzap** [comment]
https://github.com/theforeman/theforeman.org/pull/1941

**Comment by lzap** [comment]
Rebased, resolved tests hopefully.

**Review by ezr-ondrej — APPROVED** [review]
Thanks @lzap ! :+1:

**Pull request merged**